core/trino-main/src/test/java/io/trino/execution/TestAddColumnTask.java

```diff
@@ -0,0 +1,179 @@
+
+        getFutureValue(executeAddColumn(asQualifiedName(tableName), new Identifier("new_col"), INTEGER, Optional.of("test comment"), false, false));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(
+                        new ColumnMetadata("test", BIGINT),
+                        ColumnMetadata.builder()
+                                .setName("new_col")
+                                .setType(INTEGER)
+                                .setComment(Optional.of("test comment"))
+                                .build()));
+    }
+
+    @Test
+    public void testAddColumnWithColumnProperty()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+        Property columnProperty = new Property(new Identifier("column_property"), new LongLiteral("111"));
+
+        getFutureValue(executeAddColumn(asQualifiedName(tableName), new Identifier("new_col"), INTEGER, ImmutableList.of(columnProperty), false, false));
+        ColumnMetadata columnMetadata = metadata.getTableMetadata(testSession, table).getColumns().stream()
+                .filter(column -> column.getName().equals("new_col"))
+                .collect(onlyElement());
+        assertThat(columnMetadata.getProperties()).containsExactly(Map.entry("column_property", 111L));
+    }
+
+    @Test
+    public void testAddColumnNotExistingTable()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("not_existing_table");
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeAddColumn(asQualifiedName(tableName), new Identifier("test"), INTEGER, Optional.empty(), false, false)))
+                .hasErrorCode(TABLE_NOT_FOUND)
+                .hasMessageContaining("Table '%s' does not exist", tableName);
+    }
+
+    @Test
+    public void testAddColumnNotExistingTableIfExists()
+    {
+        QualifiedName tableName = qualifiedName("not_existing_table");
+
+        getFutureValue(executeAddColumn(tableName, new Identifier("test"), INTEGER, Optional.empty(), true, false));
+        // no exception
+    }
+
+    @Test
+    public void testAddColumnNotExists()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+
+        getFutureValue(executeAddColumn(asQualifiedName(tableName), new Identifier("test"), INTEGER, Optional.empty(), false, true));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("test", BIGINT)));
```

Comment:
This can be avoided - as Trino shouldn't throw an exception if we add a duplication column with `IF NOT EXISTS`